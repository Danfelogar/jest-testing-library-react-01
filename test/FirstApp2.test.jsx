import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Testing in <FirstApp />", () => {
  const title = "Hola soy el titulo";
  const subtitle = "I am a subtitle";

  test("should match with the snapshot", () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test("should show a title `Hola soy el titulo`", () => {
    render(<FirstApp title={title} />);

    //screen is the object that contains the render and allow us to interact with the component
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("should show at title in h1", () => {
    render(<FirstApp title={title} />);

    //screen is the object that contains the render and allow us to interact with the component
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('should show subtitle "I am a subtitle"', () => {
    render(<FirstApp title={title} subTitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
