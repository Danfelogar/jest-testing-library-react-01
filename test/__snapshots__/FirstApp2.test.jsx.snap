// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <FirstApp /> should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy el titulo
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
